bentoml/yatai/yatai_service_impl.py

```
@@ -578,6 +588,101 @@ def ContainerizeBento(self, request, context=None):
 
-            # pylint: enable=unused-argument
+        def UploadBento(self, request_iterator, context=None):
+            try:
+                with TempDirectory() as temp_dir:
+                    temp_tar_path = os.path.join(
+                        temp_dir, f'{uuid.uuid4().hex[:12]}.tar'
+                    )
+                    with open(temp_tar_path, 'wb+') as file:
+                        for request in request_iterator:
+                            if not request.bento_bundle:
+                                bento_name = request.bento_name
+                                bento_version = request.bento_version
+                            else:
+                                if (
+                                    bento_name == request.bento_name
+                                    and bento_version == request.bento_version
+                                ):
+                                    file.write(request.bento_bundle)
+                                else:
+                                    raise BentoMLException(
+                                        f"Incoming stream request doesn't match "
+                                        f"with initial request info "
+                                        f"{bento_name}:{bento_version} - "
+                                        f"{request.bento_name}:"
+                                        f"{request.bento_version}"
+                                    )
+                        bento_id = f"{bento_name}_{bento_version}"
+                        with lock(self.db, [(bento_id, LockType.READ)]) as (sess, _):
+                            bento_pb = self.db.metadata_store.get(
+                                sess, bento_name, bento_version
+                            )
+                            file.seek(0)
+                            if bento_pb:
+                                # If there is a directory exist, it will be removed
```

let's raise an exception here instead :D